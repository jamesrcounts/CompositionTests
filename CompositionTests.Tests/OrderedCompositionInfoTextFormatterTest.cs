using System.ComponentModel.Composition;
using System.ComponentModel.Composition.Hosting;
using System.IO;
using System.Reflection;
using ApprovalTests;
using Microsoft.ComponentModel.Composition.Diagnostics;
using Microsoft.VisualStudio.TestTools.UnitTesting;

namespace CompositionTests.Tests
{
    public interface IFileInfoProxy
    {
    }

    public interface IRemoveFiles
    {
        void Push(IFileInfoProxy source);
    }

    [Export(typeof(IRemoveFiles))]
    public class FileRemover : IRemoveFiles
    {
        public void Push(IFileInfoProxy source)
        {
            return;
        }
    }

    [TestClass]
    public class OrderedCompositionInfoTextFormatterTest
    {
        [TestMethod]
        public void ApproveFormattedText()
        {
            var catalog = new AssemblyCatalog(Assembly.GetExecutingAssembly());
            var container = new CompositionContainer(catalog);
            var info = new CompositionInfo(catalog, container);
            using (var output = new StringWriter())
            {
                OrderedCompositionInfoTextFormatter.Write(info, output);
                Approvals.Verify(output);
            }
        }
    }
}