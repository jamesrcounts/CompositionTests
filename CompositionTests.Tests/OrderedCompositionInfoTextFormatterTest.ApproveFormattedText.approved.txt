[Part] CompositionTests.Tests.CompositionTest+TestingPart from: AssemblyCatalog (Assembly="CompositionTests.Tests, Version=1.0.0.0, Culture=neutral, PublicKeyToken=null")
  [Export] CompositionTests.Tests.CompositionTest+TestingPart (ContractName="CompositionTests.Tests.CompositionTest+TestingPart")

[Part] CompositionTests.Tests.FileRemover from: AssemblyCatalog (Assembly="CompositionTests.Tests, Version=1.0.0.0, Culture=neutral, PublicKeyToken=null")
  [Export] CompositionTests.Tests.FileRemover (ContractName="CompositionTests.Tests.IRemoveFiles")

